src/models/model.cpp

```diff
@@ -1625,10 +1636,25 @@ void model::backward_prop(bool compute_weight_grads_only)
     }
     else {
       do_layer_backward_prop_begin_cbs(&l);
-      l.back_prop();
+      if (enable_layer)
+        l.back_prop();
       do_layer_backward_prop_end_cbs(&l);
     }
 
+    // Add parents to disabled layers if this layer is disabled
+    if (envvar_disable_layers) {
+      // Start propagating disabled layers up
+      if (l.get_backprop_requirements() == PROPAGATE_NOTHING)
+        enable_layer = false;
+
+      // Either start or continue propagating disabled layers
+      if (!enable_layer) {
+        for (auto& parent : l.get_parent_layers()) {
+          disabled_layers.insert(parent);
+        }
+      }
```

Comment:
I think this would disable all parent layers even if the parent has two children one who has backprop disabled and one who does not.

Reply:
@fiedorowicz1 Fixed.